extensions/sources/filebase/src/main/java/io/smallrye/ext/config/source/filebase/AbstractUrlBasedSource.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.ext.config.source.filebase;
+
+import io.smallrye.ext.config.events.ChangeEventNotifier;
+import io.smallrye.ext.config.source.base.EnabledConfigSource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import lombok.extern.java.Log;
+
+
+/**
+ * URL Based property files
+ * 
+ * Load some file from a file and convert to properties.
+ * @author <a href="mailto:phillip.kruger@redhat.com">Phillip Kruger</a>
+ */
+@Log
+public abstract class AbstractUrlBasedSource extends EnabledConfigSource implements Reloadable {
+    private final LinkedHashMap<URL,Map<String, String>> propertiesMap = new LinkedHashMap<>();
+    private final Map<String, String> properties = new HashMap<>();
+    private String urlInputString;
```

Comment:
This is a lot of mutable state, which is a departure from the existing implementations.

Reply:
made the variable final